src/TitleForm/TitleForm.test.js

```diff
@@ -73,6 +73,35 @@ describe('TitleForm', () => {
     expect(claimingIntervalField).toHaveValue(null);
   });
 
+  it('should validate \'Claiming interval\' field', async () => {
```

Comment:
I suggest separating all these checks into separate test cases. You have a lot of checks here, if one of them fails, you won't be able to detect which one exactly does. Try the next structure:
```javascript
describe('when "Claiming active" unchecked', () => {
  it('field shouldn't be required')
});

describe('when "Claiming active" checked', () => {
  it('field should be required')
});
and so on
```

Reply:
Thanks for the suggestion! However, this test is also designed to track the state of the field over time, so it is reduced to one test, this allows us to ensure that validation works when dependent fields change.